packages/core/src/State.ts

```diff
@@ -316,7 +321,9 @@ export class State<
    * Whether the current state configuration has a state node with the specified `tag`.
    * @param tag
    */
-  public hasTag(tag: string): boolean {
+  public hasTag(
+    tag: TMeta extends IsTypegenActive ? TMeta['tags'] : string
```

Comment:
Nit: maybe we can condense this with a special type:

```js
tag: WithTypegen<TMeta, 'tags', string>
```

Reply:
Having fiddled around with implementing it, I think this:

![image](https://user-images.githubusercontent.com/28293365/133398062-b78411c1-8c79-4b73-9291-7083f0404541.png)

reads better than this:

![image](https://user-images.githubusercontent.com/28293365/133398111-f59a5028-edc7-470e-baf1-90b92a67ed81.png)